source/dmocks/mocks.d

```diff
@@ -406,6 +440,134 @@ public class ExpectationSetup
     }
 }
 
+public class TemplatedExpectationSetup(T)
+{
+    import dmocks.arguments;
+    import dmocks.expectation;
+    import dmocks.dynamic;
+    import dmocks.qualifiers;
+    import dmocks.call;
+
+    private alias Ret = ReturnType!T;
+
+    private alias Params = Parameters!T;
+
+    private CallExpectation _expectation;
+
+    private Call _setUpCall;
+
+    this(CallExpectation expectation, Call setUpCall)
+    {
+        assert(expectation !is null, "can't create an ExpectationSetup if expectation is null");
```

Comment:
Use `in (expectation !is null, "can't create an ExpectationSetup if expectation is null")` instead?

Reply:
done